### PR TITLE
Refactor search bar & filters to conditionally render new look with application header

### PR DESCRIPTION
**Pull request by zhongnansu**
### Description

- Conditionally render new filter with new application header behind the toggle
- hide filter row, when there's no filters applied
- move save query into the filter menu
- add additional "Add filter" into menu
- add additional "Save query" into menu
- move combined menu to be in front of the search bar

## Screenshot

- By default, with "newHome" toggle turned off, search bar & filter bar UI remains the same as what currently is.

https://github.com/user-attachments/assets/963337dc-060c-48a1-b582-f3fdfbcbc17a

- With newHome toggle turned on, new application header will show up, along with the new look of filter bar & search bar menu options

https://github.com/user-attachments/assets/68610ec9-27a7-4134-9a92-c3ec26e68a8c

## Changelog

- refactor: Refactor search bar & filters to conditionally render new look with application header

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/data/public/ui/filter_bar/_global_filter_group.scss
Comment (virajsanghvi) [inline src/plugins/data/public/ui/filter_bar/_global_filter_group.scss:65]: can we use $euiSize variables? why 10 btw?

File: src/plugins/data/public/ui/filter_bar/_global_filter_group.scss
Comment (virajsanghvi) [inline src/plugins/data/public/ui/filter_bar/_global_filter_group.scss:59]: not sure what element this class is on, but does that element have a component prop we can set to get danger styling?

File: src/plugins/data/public/ui/filter_bar/filter_bar.tsx
Comment (virajsanghvi) [inline src/plugins/data/public/ui/filter_bar/filter_bar.tsx:184]: should this be a localized string?

File: src/plugins/data/public/ui/saved_query_management/saved_query_management_component.tsx
Comment (virajsanghvi) [inline src/plugins/data/public/ui/saved_query_management/saved_query_management_component.tsx:72]: is this changing the default state of the popover to true? it should start closed, no?

or is this even used anymore? do we want to remove?

File: src/plugins/data/public/ui/saved_query_management/saved_query_management_component.tsx
Comment (virajsanghvi) [inline src/plugins/data/public/ui/saved_query_management/saved_query_management_component.tsx:212]: Was this taken out of a popover? If so, why are we still using popover components inside?

Judging from video, is this just replacing content of current context menu?

File: src/plugins/data/public/ui/filter_bar/_global_filter_group.scss
Comment (zhongnansu) [reply src/plugins/data/public/ui/filter_bar/_global_filter_group.scss:65]: Updated to use $euiSizeS (8px)

File: src/plugins/data/public/ui/filter_bar/_global_filter_group.scss
Comment (zhongnansu) [reply src/plugins/data/public/ui/filter_bar/_global_filter_group.scss:59]: It's used in the one of the filter options "remove all"
image

The element prop is `EuiContextMenuItemProps`, which doesn't comes with a styling prop.
image

File: src/plugins/data/public/ui/filter_bar/filter_bar.tsx
Comment (zhongnansu) [reply src/plugins/data/public/ui/filter_bar/filter_bar.tsx:184]: yes, will update to use i18n

File: src/plugins/data/public/ui/saved_query_management/saved_query_management_component.tsx
Comment (zhongnansu) [reply src/plugins/data/public/ui/saved_query_management/saved_query_management_component.tsx:72]: Yes, it used to control the popover, but since the refactor separates the popover and the component that renders inside it(filter options, or save query panel, or add filter panel), this state is no longer needed here. I'll remove it.

File: src/plugins/data/public/ui/saved_query_management/saved_query_management_component.tsx
Comment (zhongnansu) [reply src/plugins/data/public/ui/saved_query_management/saved_query_management_component.tsx:212]: yes, instead of only rendering the filter menu options, this filter popover will now need to render 3 components, filter options, save query panel, and add filter panel. This separation is to make things easier. It's replacing the current context menu, I leave the other popover components unchanged, so that when the toggle is off, the existing experiences remains the same

File: src/plugins/data/public/ui/filter_bar/filter_options.tsx
Comment (virajsanghvi) [inline src/plugins/data/public/ui/filter_bar/filter_options.tsx:315]: Is there any value to defining this here instead of within your case statement below?

File: src/plugins/data/public/ui/saved_query_management/saved_query_management_component.tsx
Comment (virajsanghvi) [reply src/plugins/data/public/ui/saved_query_management/saved_query_management_component.tsx:212]: I missed filter_options.tsx before i think which would have helped make sense of this.

**Review by virajsanghvi — DISMISSED** [review]

**Comment by Flyingliuhub** [comment]
please make sure that Changelog before you merge, thanks

**Review by Flyingliuhub — APPROVED** [review]

**Review by Flyingliuhub — APPROVED** [review]

File: src/plugins/data/public/ui/filter_bar/_global_filter_group.scss
Comment (AMoo-Miki) [inline src/plugins/data/public/ui/filter_bar/_global_filter_group.scss:59]: Can you please create an issue on OUI to allow setting the `color` in OuiContextItem?

**Review by virajsanghvi — APPROVED** [review]
Approving assuming checks pass

File: src/plugins/data/public/ui/filter_bar/_global_filter_group.scss
Comment (zhongnansu) [reply src/plugins/data/public/ui/filter_bar/_global_filter_group.scss:59]: https://github.com/opensearch-project/oui/issues/1345

**Pull request merged**
